src/test/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandlerTest.java

```diff
@@ -278,6 +337,76 @@ public void shouldUpdateItemWithNewStatusIfHoldingAlreadyInPayload()
     Assert.assertNotNull(eventPayload.getContext().get(HOLDINGS.value()));
   }
 
+  @Test
+  public void shouldUpdateItemWithRetryAndRemoveRetryCounterFromPayload()
+    throws UnsupportedEncodingException, InterruptedException, ExecutionException, TimeoutException {
+
+    String itemId = UUID.randomUUID().toString();
+    String holdingId = UUID.randomUUID().toString();
+    String materialTypeId = UUID.randomUUID().toString();
+    String permanentLoanTypeId = UUID.randomUUID().toString();
+    JsonObject metadata = new JsonObject();
+    String instanceId = String.valueOf(UUID.randomUUID());
+    String hrid = UUID.randomUUID().toString();
+    String permanentLocationId = UUID.randomUUID().toString();
+
+    Item actualItem = new Item(itemId, "2", holdingId, "test", new Status(AVAILABLE), materialTypeId, permanentLoanTypeId, metadata);
+
+    when(fakeReaderFactory.createReader()).thenReturn(fakeReader);
+    when(fakeReader.read(any(MappingRule.class))).thenReturn(StringValue.of(IN_PROCESS.value()));
+    when(mockedStorage.getItemCollection(ArgumentMatchers.any(Context.class))).thenReturn(mockedItemCollection);
+    when(mockedStorage.getHoldingsRecordCollection(ArgumentMatchers.any(Context.class))).thenReturn(mockedHoldingsCollection);
+    when(mockedItemCollection.findById(anyString())).thenReturn(CompletableFuture.completedFuture(actualItem));
+
+    doAnswer(invocationOnMock -> {
+      HoldingsRecord holdings = new HoldingsRecord()
+        .withId(holdingId)
+        .withHrid(hrid)
+        .withInstanceId(instanceId)
+        .withPermanentLocationId(permanentLocationId)
+        .withVersion(1);
+      Consumer<Success<HoldingsRecord>> successHandler = invocationOnMock.getArgument(1);
+      successHandler.accept(new Success<>(holdings));
+      return null;
+    }).when(mockedHoldingsCollection).findById(anyString(),any(Consumer.class),any(Consumer.class));
+
+    doAnswer(invocationOnMock -> {
+      Consumer<Failure> failureHandler = invocationOnMock.getArgument(2);
+      failureHandler.accept(new Failure("", 409));
+      return null;
+    }).doAnswer(invocationOnMock -> {
+      Item itemRecord = invocationOnMock.getArgument(0);
+      Consumer<Success<Item>> successHandler = invocationOnMock.getArgument(1);
+      successHandler.accept(new Success<>(itemRecord));
+      return null;
+    }).when(mockedItemCollection).update(any(), any(), any());
+
+    MappingManager.registerReaderFactory(fakeReaderFactory);
+    MappingManager.registerWriterFactory(new ItemWriterFactory());
+
+    Item itemRecord = new Item(itemId, "2", holdingId, new Status(AVAILABLE), materialTypeId, permanentLoanTypeId, metadata);
+
+    Record record = new Record().withParsedRecord(new ParsedRecord().withContent(PARSED_CONTENT));
+    HashMap<String, String> context = new HashMap<>();
+    context.put(ITEM.value(), Json.encode(itemRecord));
```

Comment:
```suggestion
    context.put(ITEM.value(), Json.encode(itemRecord));
```
context.put(ITEM.value(), ItemUtil.mapToMappingResultRepresentation(itemRecord));

please try to apply ItemUtil.mapToMappingResultRepresentation() method so that the itemRecord corresponds to the inventory-storage schema and will pass validation

Reply:
Thank you, corrected.